### PR TITLE
fix: re-gen static resources from Helm chart and update doc for doing it

### PR DESCRIPTION
**Pull request by erikgb**
## Description

This regenerates the static resources for deploying the operator and updates the docs for doing it. Also adds a script to update the static resources (both multi-file and single-file).

Also includes som minor changes/additions to the Helm chart to make the Helm chart and static resources equivalent:

- Removed the conditional around RBAC for leases (replicas > 1). To my knowledge, it is common to add RBAC supporting leader election even with leader elections disabled (single replica).

## Related issues
- Close #XXX

Remove this section if you don't have related PRs.

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by josedonizetti** [comment]
@erikgb can we get hid of `hack/update-starboard.yaml.sh` on this PR as well? It should be only one action to update all static, including the single-file.

**Comment by erikgb** [comment]
> @erikgb can we get hid of `hack/update-starboard.yaml.sh` on this PR as well? It should be only one action to update all static, including the single-file.

Yes, sure! The two scripts can be merged. I'll fix it tomorrow morning.

**Comment by erikgb** [comment]
> @erikgb can we get hid of `hack/update-starboard.yaml.sh` on this PR as well? It should be only one action to update all static, including the single-file.

@josedonizetti Done, PTAL!

**Review by JAORMX — APPROVED** [review]

File: deploy/helm/templates/deployment.yaml [hunk 1]
```diff
@@ -71,6 +73,8 @@ spec:
               value: {{ .Values.operator.batchDeleteDelay | quote }}
             - name: OPERATOR_METRICS_BIND_ADDRESS
               value: ":8080"
+            - name: OPERATOR_METRICS_FINDINGS_ENABLED
+              value: {{ .Values.operator.metricsFindingsEnabled | quote }}
```
Comment (chen-keinan) [inline deploy/helm/templates/deployment.yaml:77]: I would make it in a separate PR , to not mix concerns , also update pr description

**Review by chen-keinan — APPROVED** [review]

File: deploy/helm/templates/deployment.yaml (hunk 1 above)
Comment (erikgb) [reply deploy/helm/templates/deployment.yaml:77]: Ideally I agree with you, but there are also other minor diffs fixed in this PR.

Comment (chen-keinan) [reply deploy/helm/templates/deployment.yaml:77]: because `OPERATOR_METRICS_FINDINGS_ENABLED` is related to metrics feature and If for some reason we will need to revert this PR commit we should revert the metrics features flag

Comment (erikgb) [reply deploy/helm/templates/deployment.yaml:77]: https://github.com/aquasecurity/trivy-operator/pull/179. I'll rebase this PR after merging the other one.

Comment (erikgb) [reply deploy/helm/templates/deployment.yaml]: Rebased

**Review by josedonizetti — APPROVED** [review]

**Comment by chen-keinan** [comment]
LTGM :rocket:
